mm2src/coins/tendermint/tendermint_tx_history_v2.rs

```diff
@@ -0,0 +1,754 @@
+                internal_id: args.internal_id,
+                timestamp: common::now_ms() / 1000,
+                kmd_rewards: None,
+                transaction_type: Default::default(),
+            }))
+        }
+
+        async fn fetch_and_insert_txs<Coin, Storage>(
+            coin: &Coin,
+            storage: &Storage,
+            query: String,
+            active_assets: &HashMap<String, String>,
+        ) -> Result<(), Stopped<Coin, Storage>>
+        where
+            Coin: TendermintTxHistoryOps,
+            Storage: TxHistoryStorage,
+        {
+            let mut page = 1;
+            let mut iterate_more = true;
+
+            let client = try_or_return_stopped_as_err!(
+                coin.get_rpc_client().await,
+                StopReason::RpcClient,
+                "could not get rpc client"
+            );
+            while iterate_more {
+                let response = try_or_return_stopped_as_err!(
+                    client
+                        .perform(TxSearchRequest::new(
+                            query.clone(),
+                            false,
+                            page,
+                            TX_PAGE_SIZE,
+                            TendermintResultOrder::Ascending.into(),
+                        ))
+                        .await,
+                    StopReason::RpcClient,
+                    "tx search rpc call failed"
+                );
+
+                let mut tx_details: HashMap<String, Vec<TransactionDetails>> = HashMap::new();
+                for tx in response.txs {
+                    let internal_id = H256::from(tx.hash.as_bytes()).reversed().to_vec().into();
+
+                    let deserialized_tx = try_or_return_stopped_as_err!(
+                        cosmrs::Tx::from_bytes(tx.tx.as_bytes()),
+                        StopReason::Marshaling,
+                        "Could not deserialize transaction"
+                    );
+
+                    let fee_details = try_or_return_stopped_as_err!(
+                        get_fee_details(deserialized_tx.auth_info.fee, coin),
+                        StopReason::Marshaling,
+                        "get_fee_details failed"
+                    );
+
+                    let msg = try_or_return_stopped_as_err!(
+                        deserialized_tx.body.messages.first().ok_or("Tx body couldn't be read."),
+                        StopReason::Marshaling,
+                        "Tx body messages is empty"
+                    );
+
+                    // If we don't get expected type_url, then continue.
+                    // (This check exists for not allocating ParseTxDetailsArgs when we don't use
+                    // it. We could use it in match statement but that way we have to duplicate ~10
+                    // lines of code for each pattern.)
+                    if ![SEND_TYPE_URL, CLAIM_HTLC_TYPE_URL, CREATE_HTLC_TYPE_URL].contains(&msg.type_url.as_str()) {
```

Comment:
Do you know, does the compiler optimize this code not to allocate this static array each time the function is called?

Reply:
It's one time allocated on ro data